google/stackdriver_monitoring/spectator_client.py

```diff
@@ -178,12 +242,31 @@ def collect_metrics(self, host, port, params=None):
     url = 'http://{host}:{port}/spectator/metrics{query}'.format(
         host=host, port=port, query=query)
     response = urllib2.urlopen(url)
+
     all_metrics = json.JSONDecoder(encoding='utf-8').decode(response.read())
+    try:
+      self.__log_scan_diff(host, port, all_metrics.get('metrics', {}))
```

Comment:
Seems like control flow here to call this only if `--log_metric_diff` would be clearer than calling this each time and setting internal object state based on the flag.